### PR TITLE
statistics: add refresher

### PR DESCRIPTION
**Pull request by hi-rustin**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/50132

Problem Summary:

### What changed and how does it work?

- Added a refresher to help us build the queue.
- Added an IsValidToAnalyze function to check if it is valid before we do the job.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by hi-rustin — COMMENTED** [review]
/self-check

File: pkg/statistics/handle/autoanalyze/refresher/refresher.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+	defer r.statsHandle.SPool().Put(se)
+	sctx := se.(sessionctx.Context)
+	// Pick the table with the highest weight.
+	for r.jobs.Len() > 0 {
+		job := heap.Pop(r.jobs).(*priorityqueue.TableAnalysisJob)
+		if !job.IsValidToAnalyze(
+			sctx,
+		) {
+			statslogutil.StatsLogger().Info(
+				"Table is not ready to analyze",
```
Comment (qw4990) [inline pkg/statistics/handle/autoanalyze/refresher/refresher.go:66]: Should we also log the reason why it is not ready to improve the diagnosibility?

Comment (hi-rustin) [reply pkg/statistics/handle/autoanalyze/refresher/refresher.go:66]: I think we already logged it in the IsValidToAnalyze function.

File: pkg/statistics/handle/autoanalyze/priorityqueue/job.go [hunk 2]
```diff
@@ -40,6 +42,88 @@ type TableAnalysisJob struct {
...
+		return false
+	}
+
+	// Failed analysis duration is less than 2 times the average analysis duration.
+	// Skip this table to avoid too much failed analysis.
+	onlyFailedAnalysis := lastFailedAnalysisDuration != noRecord && averageAnalysisDuration == noRecord
+	meetSkipCondition := lastFailedAnalysisDuration != noRecord &&
+		lastFailedAnalysisDuration < 2*averageAnalysisDuration
+	if lastFailedAnalysisDuration == justFailed || onlyFailedAnalysis || meetSkipCondition {
+		statslogutil.StatsLogger().Info(
```
Comment (qw4990) [inline pkg/statistics/handle/autoanalyze/priorityqueue/job.go:113]: How about returning these reasons or errors to the caller?
```
ok, reason := IsValidToAnalyze(...)
if !ok {
    statslogutil.StatsLogger().Info("Table is not ready to analyze", job, reason)
}
```

**Review by qw4990 — APPROVED** [review]
The reset LGTM

**Review by hawkingrei — APPROVED** [review]

**Review by qw4990 — APPROVED** [review]

**Pull request merged**